### DISCUSSION TITLE
[QUESTION] Can I run two or more connection calculations at the same time using the API?

### DISCUSSION THREAD

**Discussion by verwers** [idea-statica/ideastatica-public#151] [General Discussion]
I am interested in parallel pooling the changed parameters in a IDEA StatiCa connection to speed up an iteration. However, I am not sure how IDEA Statica behaves if I open multiple instances.

**Comment by mpospisil** [comment]
Hi Steven,

I discussed this issue with Nathan today. [IdeaStatiCa.ConnectionClient](https://github.com/idea-statica/ideastatica-public/tree/main/src/Examples/ConnHiddenCalc/IdeaStatiCa.ConnectionClient)  should be thread safe - you can create more instances and open more ideacon files same time. You should be able to run more CBFEM caluculations in parallel but it will probably have an impact on the performance because our FEM solver is multithreaded. The soultion could be to provide our ConnectionHiddenCalculator as a web service and use horizontal scalling in our cloud (but we only discuss this option). If you have any technical issues we are open for discussion and I'm sure we will find a good solution.

Martin
